src/server/logger.js

```diff
@@ -1,27 +1,164 @@
 'use strict';
 
-var levels = ['trace', 'info', 'log', 'debug', 'warn', 'error'],
-    debug = require('debug');
+const chalk = require('chalk'),
+    moment = require('moment'),
+    ColorHash = require('color-hash');
+
+// NOP color option to use default
+const nullcolor = (output) => output;
+
+// Stores logging levels and associated output colors
+const LEVELS = {
+    'trace': {
+        bg: nullcolor,
+        fg: nullcolor
+    }, 
+    'info': {
+        bg: nullcolor,
+        fg: nullcolor
+    },
+    'log': {
+        bg: nullcolor,
+        fg: nullcolor
+    },
+    'debug': {
+        bg: nullcolor,
+        fg: nullcolor
+    },
+    'warn':  {
+        bg: chalk.bgYellow,
+        fg: chalk.black
+    },
+    'error': {
+        bg: chalk.bgRed,
+        fg: chalk.white
+    }
+};
 
 // set which debug levels to send to stderr, the rest will go to stdout
 const STDERR = ['warn', 'error'];
 
-var Logger = function(name) {
-    this._name = name;
-    /* eslint-disable no-console*/
-    levels.forEach(lvl => {
-        this[lvl] = debug(`${name}:${lvl}`);
-        if (STDERR.find(item => item === lvl)) {
-            this[lvl].log = console.error.bind(console);
-        } else { // send to stdout
-            this[lvl].log = console.log.bind(console);
+// Get format from env variable
+const dateformat = process.env.DEBUG_DATE || 'MM.DD HH:mm:ss',
+    useColors = (process.env.DEBUG_COLORS || true) == true;
+
+// For hashing names to colors
+const colorHash = new ColorHash({lightness: 0.75});
+
+class Logger {
+    /**
+     * Create a new logger with a given namespace
+     * @param {String} name Name/namespace for this Logger 
+     */
+    constructor(name) {
+        this._name = name;
+
+        if(this._shouldBeEnabled())
+        {
+            Object.keys(LEVELS).forEach(lvl => {
+                this[lvl] = this._log.bind(this, lvl);
+            });
+        } else {
+            Object.keys(LEVELS).forEach(lvl => {
+                this[lvl] = this._nop;
+            });
         }
-    });
-    /* eslint-enable no-console*/
-};
 
-Logger.prototype.fork = function(name) {
-    return new Logger([this._name, name].join(':'));
-};
+        // Create color from string hash if available
+        if(chalk.supportsColor.has16m === true){
```

Comment:
shouldn't this also check for the 256color mode? something like `chalk.supportsColor.has16m || chalk.supportsColor.has16m ` ?

as is it doesn't work in xterm, if you comment out the conditional it does.

Reply:
AFAIK (based on the level system they use) 16m support implies 256 color support, so I've made it just check for 256 color.